src/base/utils/fs.cpp

```diff
@@ -518,3 +518,10 @@ QString Utils::Fs::cacheLocation()
         locationDir.mkpath(locationDir.absolutePath());
     return location;
 }
+
+QString Utils::Fs::tempPath()
+{
+    static const QString path = QDir::tempPath() + "/.qBittorrent/";
+    QDir().mkdir(path);
```

Comment:
Another reason is I'm not confident to use static variable destructor to remove the folder. I mean the shutdown process is complex enough and explicitly noting the folder remove could avoid hard-to-find bugs.